### PR TITLE
Improve README and CONTRIBUTING to make the project more accessible.

### LINKED ISSUE #498: Spelling mistake in documentation
There is a n (should be a m) in the word solr_implentation, the first line in the module example at https://monarch-app.monarchinitiative.org/#basic-example-as-a-module.

The module obviously fails to load.

### LINKED ISSUE #600: Create CONTRIBUTING.md
We need to document several aspects of how to contribute to the Monarch app. The top-level CONTRIBUTING.md should reference a separate CONTRIBUTING.md in each of the backend and frontend sections of the app. 

- [ ] Top level CONTRIBUTING.md with information about the project and setup of the development environment.
- [ ] backend: detail of setup for development, including dependencies and suggested environment.
- [ ] backend: how to pull a PR branch and test functionality, including how to set up the environment for local testing.
- [ ] frontend: detail setup for development - including dependencies, suggested environment, and how to run locally.
- [ ] frontend: how to pull a PR and test functionality
- [ ] frontend: how to run linting and testing framework locally.

### PR DESCRIPTION
**Pull request by amc-corey-cox**
### Related issues

- Closes #600 

### Summary

- adds CONTRIBUTING.md for main repository, frontend, and backend
- changes README.md for main repository, frontend, and backend
- improves documentation and adds useability improvement for local usage, development, and testing.

### Checks

- [ ] All tests have passed (or issues created for failing tests)

### CODE REVIEW

**Comment by amc-corey-cox** [comment]
This isn't fully cooked yet but I would like to have peoples opinions especially on the Code of Conduct and top-level README project description and usage.

File: .python-version
Comment (glass-ships) [inline .python-version:1]: just curious, what's this file for?

Comment (amc-corey-cox) [reply .python-version:1]: It's used by pyenv to manage the python version dynamically. It ensures anyone using pyenv will be using the correct version. I can go ahead and add it to .gitignore but it is kind of convenient and anyone who knows what it is will automatically know what python version to use for development.

**Comment by glass-ships** [comment]
I think my only other comment would be that some content feels very duplicated (like useful links, requirements, etc.) which may not be entirely necessary, and can possibly cause headaches for us in the future if we need to update things. 
Otherwise looking great though! Thanks for all your hard work on this

**Comment by amc-corey-cox** [comment]
Yes, parts of it are a little repetitive. I understand the concern about keeping it properly maintained. The problem that I see is we shouldn't count on someone looking through all of the documentation for what they want. There is probably a better way to manage this but I do think we need to have the bare-bones of what someone wants to do right in front of them. We can refer to other resources for anything more complicated. I'm trying to do that but I'm sure I'm duplicating more than I need to.

**Comment by glass-ships** [comment]
Yeah, and that totally makes sense. Barebones plus directing to more specific and thorough documentation is a good way to go about it, and it's definitely something we can iterate over too as we feel the need to. What you've got going is great for a first pass!

**Comment by amc-corey-cox** [comment]
This could probably be ready to merge in but there is one more thing I would really like to properly capture. How to build/host your own dataset. I feel like this is the main important use-case for someone who might want to pull down our whole stack and play with it locally. 

If someone wanted to build a test KG (i.e. using monarch-ingest plus their favorite changes) and use the Monarch Web App to query it locally on their machine to see how it performs. This has a lot of utility for us and is something I could see interested researchers (or our graduate students) wanting to do to try out new things.

**Comment by kevinschaper** [comment]
It's awkwardly possible by forking right now but not really nicely built for that yet. 

I think the next direction we're going in will be modular ingests so that we can more easily build KGs that are plus and minus sources or mappings. 

That said, documenting how to do it now would be a good exercise for seeing where we want to go.

File: frontend/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+This CONTRIBUTING.md is a javascript/typescript web app using Bun as a replacement for Node.js (a non-browser JavaScript runtime environment) with the Vite/Vue framework. You can use the monarch-app frontend to query the Monarch KG using the web app locally.
+
+Without additional setup this will use the same FastAPI backend and data source as the website. This is primarily for front-end development of the website UI. If you want to use a local data store review the monarch-app README.md and CONTRIBUTING.md in the top level directory of this repository to understand how to manage the whole stack locally.
+
+# Useful Links
+
+- [Monarch App Website](https://next.monarchinitiative.org/) - The Monarch Initiative website, a Vue webapp accessing `monarch-py`, a Python library for interacting with the Monarch Initiative knowledge graph, which includes an optional FastAPI module that serves as the website's backend.
```
Comment (kevinschaper) [inline frontend/CONTRIBUTING.md:11]: We should just use https://monarchinitiative.org for the URL now

File: frontend/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+If you don't have an existing Node.js implementation and want to install Bun globally on your system (requires curl):
+
+```
+curl -fsSL https://bun.sh/install | bash
+```
+
+## Run using Monarch Cloud hosted API
+
+To run the frontend using the Monarch Cloud hosted API, you can use the following commands:
```
Comment (kevinschaper) [inline frontend/CONTRIBUTING.md:60]: Might need to explicitly mention being in the frontend directory  (or imply it with a little fake prompt?)

Comment (kevinschaper) [reply frontend/CONTRIBUTING.md:60]: ohh, never mind, this is the readme within the frontend directory!

**Review by kevinschaper — APPROVED** [review]
thank you! this is a huge improvement to the repo!

**Pull request merged**